### PR TITLE
[sapmexporter] Split traces before enqueuing, every request one token

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1495

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: exporter/sapmexporter/exporter.go [hunk 1]
```diff
@@ -74,64 +83,80 @@ func newSAPMTraceExporter(cfg *Config, params component.ExporterCreateParams) (c
...
-			droppedSpansCount += trace.SpanCount()
-			err = consumererror.Permanent(translateErr)
-			continue
+func (se *sapmExporter) pushTraceData(ctx context.Context, td pdata.Traces) (int, error) {
+	rss := td.ResourceSpans()
+	if rss.Len() == 0 {
+		return 0, nil
+	}
+
+	accessToken := se.retrieveAccessToken(rss.At(0))
```
Comment (tigrannajaryan) [inline exporter/sapmexporter/exporter.go:110]: Please add a comment to explain why the access token of element at index 0 can be used for all.

File: exporter/sapmexporter/exporter.go [hunk 2]
```diff
@@ -74,64 +83,80 @@ func newSAPMTraceExporter(cfg *Config, params component.ExporterCreateParams) (c
...
-			}
-			droppedSpansCount += trace.SpanCount()
+	attrs := md.Resource().Attributes()
+	if accessToken, ok := attrs.Get(splunk.SFxAccessTokenLabel); ok {
+		return accessToken.StringVal()
+	}
+	return ""
+}
+
+func filterToken(batches []*model.Batch) {
```
Comment (tigrannajaryan) [inline exporter/sapmexporter/exporter.go:142]: Function name and purpose is not clear.

File: exporter/sapmexporter/exporter.go [hunk 3]
```diff
@@ -74,64 +83,80 @@ func newSAPMTraceExporter(cfg *Config, params component.ExporterCreateParams) (c
...
+		return 0, nil
+	}
+
+	accessToken := se.retrieveAccessToken(rss.At(0))
+	batches, err := jaeger.InternalTracesToJaegerProto(td)
+	if err != nil {
+		return td.SpanCount(), consumererror.Permanent(err)
+	}
+
+	filterToken(batches)
```
Comment (tigrannajaryan) [inline exporter/sapmexporter/exporter.go:116]: Not clear what this does.

File: exporter/sapmexporter/exporter.go (hunk 1 above)
Comment (bogdandrutu) [reply exporter/sapmexporter/exporter.go:111]: Done.

File: exporter/sapmexporter/exporter.go (hunk 2 above)
Comment (bogdandrutu) [reply exporter/sapmexporter/exporter.go:146]: Done.

File: exporter/sapmexporter/exporter.go (hunk 3 above)
Comment (bogdandrutu) [reply exporter/sapmexporter/exporter.go:119]: Done.

**Review by tigrannajaryan — APPROVED** [review]
LGTM

**Pull request merged**